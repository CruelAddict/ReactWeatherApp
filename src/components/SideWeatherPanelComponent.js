import React from 'react'
import WeatherValuesList from './WeatherValuesList'

export default (props) => {
    let cityNotFound = props.state.errorMessage === 'Город не найден';
    if (props.state.isLoading) return <div className={"side-weather-panel-container"}>
        <div className={"side-weather-panel-main-data"}>
            <div className={"side-city-name-container"}>
                <h2>{props.state.city}</h2>
            </div>
            <div className={"side-temperature-container"}>
            </div>
            <div className={"side-icon-cell"}>
                <div>
                </div>
            </div>
            <div className={"delete-favorite-button-container"}>
                <button
                    onClick={props.handleFavoriteRemoval}
                    className={"delete-favorite favorites-action-button"}>×
                </button>
            </div>
        </div>
        <div className={"loading-side"}>
            <h4 className={"loading-title-side"} >Подождите, данные загружаются</h4>
            <div className={"spinner-container"}>
                <i className={"fa fa-refresh fa-spin fa-4x"}></i>
            </div>
        </div>
    </div>;
    else if(props.state.isFailed) return <div
        className={cityNotFound ? "side-weather-panel-container city-not-found" : "side-weather-panel-container"}
    >
        <div className={"side-weather-panel-main-data"}>
            <div className={"side-city-name-container"}>
                <h2>{props.state.city}</h2>
            </div>
            <div className={"side-temperature-container"}>
            </div>
            <div className={"side-icon-cell"}>
                <div>
                </div>
            </div>
            <div className={"delete-favorite-button-container"}>
                <button
                    onClick={props.handleFavoriteRemoval}
                    className={"delete-favorite favorites-action-button"}>×
                </button>
            </div>
        </div>
        {
            cityNotFound ?
            <div className={"loading-side"}>
                <h4 className={"loading-title-side side-error-message"} >{props.state.errorMessage}</h4>
            </div> :
            <div className={"loading-side"}>
                <h4 className={"loading-title-side side-error-message"} >{props.state.errorMessage}</h4>
                <div className={"refresh-button-container"}>
                    <button
                        className={"refresh-button"}
                        onClick={props.setWeather}
                    >Попробовать снова</button>
                </div>
            </div>
        }
    </div>;
    else
        return (
            <div className={"side-weather-panel-container"}>
                <div className={"side-weather-panel-main-data"}>
                    <div className={"side-city-name-container"}>
                        <h2>{props.state.city}</h2>
                    </div>
                    <div className={"side-temperature-container"}>
                        <span>{props.state.weather_data.temperature.toFixed(0)}°С</span>
                    </div>
                    <div className={"side-icon-cell"}>
                        <div className={"side-icon-container"}>
                            <img src={props.state.weather_data.icon} alt={"weather icon"}/>
                        </div>
                    </div>
                    <div className={"delete-favorite-button-container"}>
                        <button
                            onClick={props.handleFavoriteRemoval}
                            className={"delete-favorite favorites-action-button"}>×
                        </button>
                    </div>
                </div>
                <div className={"side-weather-items"}>
                    <WeatherValuesList state={props.state}/>
                </div>
            </div>
        )
}

