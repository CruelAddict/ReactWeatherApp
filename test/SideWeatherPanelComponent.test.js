import React from 'react';
import SideWeatherPanelComponent from '../src/components/SideWeatherPanelComponent';
import { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as Enzyme from "enzyme";
Enzyme.configure({ adapter: new Adapter() });

describe ('SideWeatherPanelComponent', () => {

    it('Renders in "loading" state', () => {
        const panel = shallow(
            <SideWeatherPanelComponent state={{isLoading: true}}/>
        );
        expect(panel).toMatchSnapshot()
    });

    it('Displays weather data for a city', () => {
        const panel = shallow(
            <SideWeatherPanelComponent state={{
                isLoading: false,
                city: "London",
                coordinates: {
                    latitude: 51.51,
                    longitude: -0.13
                },
                isFailed: false,
                weather_data: {
                    cloudiness: "overcast clouds",
                    humidity: 87,
                    icon: "blob:http://localhost:3000/983a1031-dd0b-42bb-a079-8feb35f05a10",
                    pressure: 1016,
                    temperature: 3.9700000000000273,
                    wind: 2.1
                }
            }}/>
        );
        expect(panel).toMatchSnapshot()
    });

    it('Displays an error if server does not respond', () => {
        const panel = shallow(
            <SideWeatherPanelComponent state={{
                isLoading: false,
                city: "London",
                coordinates: {
                    latitude: 51.51,
                    longitude: -0.13
                },
                isFailed: true,
                weather_data: {
                    cloudiness: "overcast clouds",
                    humidity: 87,
                    icon: "blob:http://localhost:3000/983a1031-dd0b-42bb-a079-8feb35f05a10",
                    pressure: 1016,
                    temperature: 3.9700000000000273,
                    wind: 2.1
                },
                errorMessage: "Сервер не отвечает"
            }}/>
        );
        expect(panel).toMatchSnapshot()
    });

    it('Displays an error if city was not found', () => {
        const panel = shallow(
            <SideWeatherPanelComponent state={{
                isLoading: false,
                city: "London",
                coordinates: {
                    latitude: 51.51,
                    longitude: -0.13
                },
                isFailed: true,
                weather_data: {
                    cloudiness: "overcast clouds",
                    humidity: 87,
                    icon: "blob:http://localhost:3000/983a1031-dd0b-42bb-a079-8feb35f05a10",
                    pressure: 1016,
                    temperature: 3.9700000000000273,
                    wind: 2.1
                },
                errorMessage: "Город не найден"
            }}/>
        );
        expect(panel).toMatchSnapshot()
    });


});
