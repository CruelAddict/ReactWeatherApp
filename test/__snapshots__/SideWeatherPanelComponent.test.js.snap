// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideWeatherPanelComponent Displays an error if city was not found 1`] = `
<div
  className="side-weather-panel-container city-not-found"
>
  <div
    className="side-weather-panel-main-data"
  >
    <div
      className="side-city-name-container"
    >
      <h2>
        London
      </h2>
    </div>
    <div
      className="side-temperature-container"
    />
    <div
      className="side-icon-cell"
    >
      <div />
    </div>
    <div
      className="delete-favorite-button-container"
    >
      <button
        className="delete-favorite favorites-action-button"
      >
        ×
      </button>
    </div>
  </div>
  <div
    className="loading-side"
  >
    <h4
      className="loading-title-side side-error-message"
    >
      Город не найден
    </h4>
  </div>
</div>
`;

exports[`SideWeatherPanelComponent Displays an error if server does not respond 1`] = `
<div
  className="side-weather-panel-container"
>
  <div
    className="side-weather-panel-main-data"
  >
    <div
      className="side-city-name-container"
    >
      <h2>
        London
      </h2>
    </div>
    <div
      className="side-temperature-container"
    />
    <div
      className="side-icon-cell"
    >
      <div />
    </div>
    <div
      className="delete-favorite-button-container"
    >
      <button
        className="delete-favorite favorites-action-button"
      >
        ×
      </button>
    </div>
  </div>
  <div
    className="loading-side"
  >
    <h4
      className="loading-title-side side-error-message"
    >
      Сервер не отвечает
    </h4>
    <div
      className="refresh-button-container"
    >
      <button
        className="refresh-button"
      >
        Попробовать снова
      </button>
    </div>
  </div>
</div>
`;

exports[`SideWeatherPanelComponent Displays weather data for a city 1`] = `
<div
  className="side-weather-panel-container"
>
  <div
    className="side-weather-panel-main-data"
  >
    <div
      className="side-city-name-container"
    >
      <h2>
        London
      </h2>
    </div>
    <div
      className="side-temperature-container"
    >
      <span>
        4
        °С
      </span>
    </div>
    <div
      className="side-icon-cell"
    >
      <div
        className="side-icon-container"
      >
        <img
          alt="weather icon"
          src="blob:http://localhost:3000/983a1031-dd0b-42bb-a079-8feb35f05a10"
        />
      </div>
    </div>
    <div
      className="delete-favorite-button-container"
    >
      <button
        className="delete-favorite favorites-action-button"
      >
        ×
      </button>
    </div>
  </div>
  <div
    className="side-weather-items"
  >
    <_default
      state={
        Object {
          "city": "London",
          "coordinates": Object {
            "latitude": 51.51,
            "longitude": -0.13,
          },
          "isFailed": false,
          "isLoading": false,
          "weather_data": Object {
            "cloudiness": "overcast clouds",
            "humidity": 87,
            "icon": "blob:http://localhost:3000/983a1031-dd0b-42bb-a079-8feb35f05a10",
            "pressure": 1016,
            "temperature": 3.9700000000000273,
            "wind": 2.1,
          },
        }
      }
    />
  </div>
</div>
`;

exports[`SideWeatherPanelComponent Renders in "loading" state 1`] = `
<div
  className="side-weather-panel-container"
>
  <div
    className="side-weather-panel-main-data"
  >
    <div
      className="side-city-name-container"
    >
      <h2 />
    </div>
    <div
      className="side-temperature-container"
    />
    <div
      className="side-icon-cell"
    >
      <div />
    </div>
    <div
      className="delete-favorite-button-container"
    >
      <button
        className="delete-favorite favorites-action-button"
      >
        ×
      </button>
    </div>
  </div>
  <div
    className="loading-side"
  >
    <h4
      className="loading-title-side"
    >
      Подождите, данные загружаются
    </h4>
    <div
      className="spinner-container"
    >
      <i
        className="fa fa-refresh fa-spin fa-4x"
      />
    </div>
  </div>
</div>
`;
